WorkloadManagementSystem/Service/JobManagerHandler.py

```diff
@@ -246,21 +255,22 @@ def __deleteJob( self, jobID ):
     if not result['OK']:
       gLogger.warn( 'Failed to delete job from the TaskQueue' )
 
-    # if it was the last job for the pilot, clear PilotsLogging about it
-    result = gPilotAgentsDB.getPilotsForJobID( jobID )
-    if not result['OK']:
-      return result
-    for pilot in result['Value']:
-      res = gPilotAgentsDB.getJobsForPilot( pilot['PilotID'] )
-      if not res['OK']:
-        return res
-      if not res['Value']:  # if list of jobs for pilot is empty, delete pilot and pilotslogging
-        ret = gPilotAgentsDB.deletePilot( pilot['PilotID'] )
-        if not ret['OK']:
-          return ret
-        ret = gPilotsLoggingDB.deletePilotsLogging( pilot['PilotID'] )
-        if not ret['OK']:
-          return ret
+    if enablePilotsLogging:
+      # if it was the last job for the pilot, clear PilotsLogging about it
+      result = gPilotAgentsDB.getPilotsForJobID( jobID )
+      if not result['OK']:
+        return result
+      for pilot in result['Value']:
+        res = gPilotAgentsDB.getJobsForPilot( pilot['PilotID'] )
+        if not res['OK']:
+          return res
+        if not res['Value']:  # if list of jobs for pilot is empty, delete pilot and pilotslogging
+          ret = gPilotAgentsDB.deletePilot( pilot['PilotID'] )
+          if not ret['OK']:
+            return ret
+          ret = gPilotsLoggingDB.deletePilotsLogging( pilot['PilotID'] )
+          if not ret['OK']:
+            return ret
```

Comment:
Don't you want to keep the interaction with `gPilotAgentsDB`?

Reply:
I have to use gPilotAgentDB and gPilotsLoggingDB to keep database separation on DB class level. Service handles multiple DB-classes, each DB-class handles just one database.